### PR TITLE
[ASM] Add RASP category of tests, including LFI, SQLi and SSRF

### PR DESCRIPTION
**Pull request by cbeauchesne**
## Motivation

This PR introduces a new category of appsec tests to cover the new exploit prevention feature (RASP). This first iteration of tests provides the basis for future tests, as the lack of available libraries to test with makes the introduction of new tests quite a complex endeavour.

## Changes

The following new tests have been introduced:

- tests/appsec/rasp/test_lfi.py:
  -  `Test_Lfi_UrlQuery`
  - `Test_Lfi_BodyUrlEncoded`
  - `Test_Lfi_BodyXml`
  - `Test_Lfi_BodyJson`

- tests/appsec/rasp/test_sqli.py:
  -  `Test_Sqli_UrlQuery`
  - `Test_Sqli_BodyUrlEncoded`
  - `Test_Sqli_BodyXml`
  - `Test_Sqli_BodyJson`  

- tests/appsec/rasp/test_ssrf.py:
  -  `Test_Ssrf_UrlQuery`
  - `Test_Ssrf_BodyUrlEncoded`
  - `Test_Ssrf_BodyXml`
  - `Test_Ssrf_BodyJson`  

These new tests are specifically included in the new `APPSEC_RASP` scenario.

The following new features  have also been introduced to the feature parity dashboard:
- `rasp_local_file_inclusion`
- `rasp_server_side_request_forgery`
- `rasp_sql_injection`
## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [x] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/appsec/rasp/__init__.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+def validate_rasp_attack(span, rule, parameters=None):
```
Comment (NachoEchevarria) [inline tests/appsec/rasp/__init__.py:1]: Should we check the presence of a stack_id?

**Review by NachoEchevarria — APPROVED** [review]
LGTM!

File: tests/appsec/rasp/__init__.py (hunk 1 above)
Comment (Anilm3) [reply tests/appsec/rasp/__init__.py:1]: I want to add a separate check to validate the presence of `stack_id` with the contents of the stack trace, so I skipped it for now as it's very hard to verify without a library.

File: tests/appsec/rasp/__init__.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    assert "_dd.appsec.json" in span["meta"], "'_dd.appsec.json' not found in 'meta'"
+
+    json = span["meta"]["_dd.appsec.json"]
+    assert "triggers" in json, "'triggers' not found in '_dd.appsec.json'"
+
+    triggers = json["triggers"]
+    assert len(triggers) == 1, "multiple appsec events found, only one expected"
+
+    trigger = triggers[0]
+    obtained_rule_id = trigger["rule"]["id"]
```
Comment (cbeauchesne) [inline tests/appsec/rasp/__init__.py:11]: You do nothing with this value, is it expected ?

Comment (Anilm3) [reply tests/appsec/rasp/__init__.py:11]: No, I just forgot to use it in the assertion below

Comment (cbeauchesne) [reply tests/appsec/rasp/__init__.py:11]: ok, taking care of it

File: docs/weblog/README.md [hunk 3]
```diff
@@ -421,8 +421,72 @@ It supports the following body fields:
...
+### \[GET,POST\] /rasp/lfi
+
+This endpoint is used to test for local file inclusion / path traversal attacks, consequently it must perform an operation on a file or directory, e.g. `open` with a relative path. The chosen operation must be injected with the `GET` or `POST` parameter.
+
+Query parameters and body fields required in the `GET` and `POST` method:
+- `file`: containing the string to inject on the file operation.
+
+The endpoint should support the following content types in the `POST` method:
+- `application/x-www-form-urlencoded`
+- `application/xml`
```
Comment (christophe-papazian) [inline docs/weblog/README.md:433]: xml is not standard among all tracers, maybe just keep the other two?

**Review by christophe-papazian — APPROVED** [review]

File: docs/weblog/README.md (hunk 3 above)
Comment (Anilm3) [reply docs/weblog/README.md:433]: I wrote it as **should** instead of **must** to allow this flexibility.

**Comment by cbeauchesne** [comment]
Setting the PR as draft, ping us when ready for review !

**Comment by cbeauchesne** [comment]
There is a conflict to handle, ping me when I can fix it.

File: utils/build/docker/python/fastapi/main.py [hunk 4]
```diff
@@ -96,6 +101,71 @@ async def tag_value_post(tag_value: str, status_code: int, request: Request):
...
+
+@app.get("/rasp/ssrf")
+@app.post("/rasp/ssrf")
+def rasp_ssrf(request: Request):
+    print("rasp_ssrf", repr(request), file=sys.stderr)
+    domain = None
+    if request.method == "GET":
+        domain = request.query_params.get("domain")
+    elif request.method == "POST":
+        body = await request.body()
```
Comment (robertomonteromiguel) [inline utils/build/docker/python/fastapi/main.py:143]: This line causes this error trace on the startup:

```Traceback (most recent call last):
  File "/usr/local/bin/uvicorn", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 409, in main
    run(
  File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 575, in run
    server.run()
  File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/site-packages/ddtrace/contrib/asyncio/patch.py", line 50, in traced_coro
    return await coro
  File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/usr/local/lib/python3.10/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "/usr/local/lib/python3.10/site-packages/ddtrace/internal/module.py", line 213, in _exec_module
    pre_exec_hook(self, module)
  File "/usr/local/lib/python3.10/site-packages/ddtrace/appsec/_iast/_loader.py", line 41, in _exec_iast_patched_module
    module_watchdog.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1017, in get_code
  File "<frozen importlib._bootstrap_external>", line 947, in source_to_code
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/main.py", line 143
    body = await request.body()
           ^^^^^^^^^^^^^^^^^^^^
SyntaxError: 'await' outside async function```

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
fastapi weblog doesn't start. Check the full stack trace on the PR comments

File: utils/build/docker/python/fastapi/main.py (hunk 4 above)
Comment (christophe-papazian) [reply utils/build/docker/python/fastapi/main.py:143]: fixed

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**